pkg/admission/mutator/shoot.go

```diff
@@ -55,6 +56,28 @@ func (s *shoot) Mutate(ctx context.Context, new, old client.Object) error {
 		return fmt.Errorf("wrong object type %T", new)
 	}
 
+	// skip if shoot is in restore or migration phase
+	if wasShootRescheduledToNewSeed(shoot) {
+		return nil
+	}
+
+	var oldShoot *gardencorev1beta1.Shoot
+	if old != nil {
+		oldShoot, ok = old.(*gardencorev1beta1.Shoot)
+		if !ok {
+			return fmt.Errorf("wrong object type %T", old)
+		}
+	}
+
+	if oldShoot != nil && isShootInMigrationOrRestorePhase(shoot) && !reflect.DeepEqual(oldShoot.Spec, shoot.Spec) {
```

Comment:
Q: Why do we need `!reflect.DeepEqual(oldShoot.Spec, shoot.Spec)`? Copy-paste from gardener-apiserver?